File: JavaScript/mexicanWave.js
function wave(str){
  let array = [];
  let index = 0;
  str = str.toLowerCase();
  for (let i = 0; i < str.length; i++) {
    let result = str.slice(0,index) + str[index].toUpperCase() + str.slice(index + 1);
    array.push(result);
    index = index + 1;
  }
  return array.filter(each => each !== str)
}

File: Java/MexicanWave.java
import java.util.Arrays;
import java.util.Objects;

public class MexicanWave {
    public static String[] wave(String str) {
        String[] wordsArray = str.split("");
        String[] result = new String[wordsArray.length];

        for (int i = 0; i < result.length; i++) {
            if (str.isEmpty()) {
                return new String[0];
            }
            if (wordsArray[i].equals(" ")) {
                continue;
            }
            wordsArray[i] = wordsArray[i].toUpperCase();
            String newStr = String.join("", wordsArray);
            result[i] = newStr;
            wordsArray[i] = wordsArray[i].toLowerCase();
        }

        result = Arrays.stream(result).filter(Objects::nonNull).toArray(String[]::new);
        return result;
    }

    public static void main(String[] args) {
        System.out.println(Arrays.toString(wave("Hello")));
        System.out.println(Arrays.toString(wave("Two words")));
        System.out.println(Arrays.toString(wave(" gap ")));
        System.out.println(Arrays.toString(wave("")));
    }
}
